### PR TITLE
Include SharedFx bundle in Hosting Bundle, instead of SharedFx .msi

### PR DESCRIPTION
**Pull request by wtgodbe**
Short-term fix for https://github.com/dotnet/aspnetcore/issues/36382. Add a wix project for the SharedFx bundles and reference that from the Hosting bundle, instead of the SharedFx .msi's. Also change the Hosting Bundle to ProjectReference SharedFrameworkBundle instead of having 2 ProjectReferences to SharedFrameworkLib - this ensures that by the time `ResolveProjectReferences` is done, the SharedFx .msi's are available so we can extract the Product Code from them & pass it along to the new wix project. Will need to backport this to servicing after I confirm it works with test builds.

I also removed `BundleProviderKey` from the SharedFx bundle after conversation with @joeloff to confirm it wasn't necessary.

CC @jamshedd

### CODE REVIEW

File: src/Installers/Windows/WindowsHostingBundle/WindowsHostingBundle.wixproj [hunk 1]
```diff
@@ -53,20 +54,7 @@
...
-      <DoNotHarvest>true</DoNotHarvest>
-    </ProjectReference>
-    <ProjectReference Include="..\SharedFrameworkLib\SharedFrameworkLib.wixproj">
-      <SetPlatform>Platform=x64</SetPlatform>
-      <Name>SharedFrameworkLib_x64</Name>
-      <Project>{5244BC49-2568-4701-80A6-EAB8950AB5FA}</Project>
-      <Private>True</Private>
-      <DoNotHarvest>true</DoNotHarvest>
-    </ProjectReference>
+    <ProjectReference Include="..\SharedFrameworkBundle\SharedFrameworkBundle.wixproj" />
```
Comment (dougbu) [inline src/Installers/Windows/WindowsHostingBundle/WindowsHostingBundle.wixproj:57]: Suggest this reference should have `ReferenceOutputAssembly="false"` and `Private="True"`

**Review by dougbu — COMMENTED** [review]
I defer to @joeloff for an actual approval

**Comment by wtgodbe** [comment]
It works! Test build confirms that after installing Hosting bundle 5.0.n, shared framework bundle 5.0.n+1 will uninstall the older runtime that came with the hosting bundle.

**Comment by jamshedd** [comment]
> It works! Test build confirms that after installing Hosting bundle 5.0.n, shared framework bundle 5.0.n+1 will uninstall the older runtime that came with the hosting bundle.

Hurray!

File: src/Installers/Windows/WindowsHostingBundle/WindowsHostingBundle.wixproj (hunk 1 above)
Comment (wtgodbe) [reply src/Installers/Windows/WindowsHostingBundle/WindowsHostingBundle.wixproj]: Done

**Review by joeloff — APPROVED** [review]

**Pull request merged**